docs/apis/communication.md

```diff
@@ -2,47 +2,372 @@
 
 The communication layer represents the base Tangle layer where so called `Messages` are gossiped around. A `Message` contains payloads and it is up to upper layers to interpret and derive functionality out of them.
 
+
 The API provides three functions to interact with this primitive layer:
-* `Data(data []byte) (string, error)`
-* `FindMessageByID(base58EncodedIDs []string) (*webapi_message.Response, error)`
-* `SendPayload(payload []byte) (string, error)`
+* [/messages/:messageID](#messagesmessageid)
+* [/messages/:messageID/metadata](#messagesmessageidmetadata)
+* [/messages/:messageID/consensus](#messagesmessageidconsensus)
+* [/data](#data)
+* [/messages/payload](#messagespayload)
+
+Client lib APIs:
+* [GetMessage()](#client-lib---getmessage)
+* [GetMessageMetadata()](#client-lib---getmessagemetadata)
+* [Data()](#client-lib---data)
+* [SendPayload()](#client-lib---sendpayload)
+
+
+##  `/messages/:messageID`
+
+Return message from the tangle.
+
+### Parameters
+
+| **Parameter**            | `messageID`      |
+|--------------------------|----------------|
+| **Required or Optional** | required       |
+| **Description**          | ID of a message to retrieve   |
+| **Type**                 | string         |
+
+### Examples
 
-#### Issuing a data message
-A data message is simply a `Message` containing some raw data (literally bytes). This type of message has therefore no real functionality other than that it is retrievable via `FindMessageByID`.
+#### cURL
 
-Example:
+```shell
+curl --location --request GET 'http://localhost:8080/messages/4MSkwAPzGwnjCJmTfbpW4z4GRC7HZHZNS33c2JikKXJc'
```

Comment:
keep this more general the first time you use a particular example ID. Like
curl --location --request GET 'http://localhost:8080/messages/:messageID'
where `messageID` is the ID of the message, e.g. 4MSkwAPzGwnjCJmTfbpW4z4GRC7HZHZNS33c2JikKXJc.

If there is a second example feel free to replace it there.. for example check out how we did it in line 53 and line 62 in
https://github.com/iotaledger/goshimmer/pull/1346/files